engines/python/setup/djl_python/lmi_trtllm/trtllm_async_service.py

```diff
@@ -0,0 +1,500 @@
+
+from openai.types.chat import ChatCompletionMessageParam
+
+# yapf: disable
+from tensorrt_llm.executor import CppExecutorError
+from tensorrt_llm.llmapi import LLM, BuildConfig
+from tensorrt_llm.llmapi.llm import RequestOutput
+from tensorrt_llm.serve.openai_protocol import (
+    ChatCompletionLogProbs, ChatCompletionLogProbsContent,
+    ChatCompletionNamedToolChoiceParam, ChatCompletionRequest,
+    ChatCompletionResponse, ChatCompletionResponseChoice,
+    ChatCompletionResponseStreamChoice, ChatCompletionStreamResponse,
+    ChatMessage, CompletionRequest, CompletionResponse,
+    CompletionResponseChoice, CompletionResponseStreamChoice,
+    CompletionStreamResponse, DeltaMessage, ErrorResponse, FunctionCall,
+    ModelCard, ModelList, ToolCall, UsageInfo)
+
+from djl_python.async_utils import create_non_stream_output, handle_streaming_response, ProcessedRequest
+from djl_python.inputs import Input
+from djl_python.outputs import Output
+from djl_python.properties_manager.trt_properties import TensorRtLlmProperties
+from djl_python.encode_decode import decode
+
+from .request_response_utils import trtllm_non_stream_output_formatter, trtllm_stream_output_formatter
+
+logger = logging.getLogger(__name__)
+
+
+class ConversationMessage(TypedDict):
+    role: str
+    content: str
+
+
+def parse_chat_message_content(
+        message: ChatCompletionMessageParam, ) -> List[ConversationMessage]:
+    role = message["role"]
+    content = message.get("content")
+
+    if content is None:
+        return []
+    if isinstance(content, str):
+        return [ConversationMessage(role=role, content=content)]
+
+    # for Iterable[ChatCompletionContentPartTextParam]
+    texts: List[str] = []
+    for part in content:
+        part_type = part["type"]
+        if part_type == "text":
+            text = part["text"]
+            texts.append(text)
+        else:
+            raise NotImplementedError(f"{part_type} is not supported")
+
+    text_prompt = "\n".join(texts)
+    return [ConversationMessage(role=role, content=text_prompt)]
+
+
+class TensorRTLlmAsyncService:
+
+    def __init__(self):
+        self.hf_configs = None
+        self.llm = None
+        self.tokenizer = None
+        self.model_name = None
+        self.initialized = False
+
+    def initialize(self, properties: dict):
+        self.hf_configs = TensorRtLlmProperties(**properties)
+        # TODO: Support all the LLM args here, and in AOT for compilation
+        self.llm = LLM(
+            self.hf_configs.model_id_or_path,
+            tokenizer=self.hf_configs.model_id_or_path,
+            tensor_parallel_size=self.hf_configs.tensor_parallel_degree,
+            trust_remote_code=self.hf_configs.trust_remote_code,
+            build_config=BuildConfig(
+                max_batch_size=self.hf_configs.max_rolling_batch_size,
+                output_timing_cache="/tmp/model.cache",
+            ),
+            **self.hf_configs.get_extra_kwargs(),
```

Comment:
1. We need to convert datatype, properties are all str value, need to convert to int and bool, we might need to declare all those fields
2. `get_extra_kwargs()` is not a `dict` this doesn't work.

Reply:
yea, that's a good point. I was hoping we can avoid the declaration of all the fields here, but seems like that's the only way.

Reply:
refactored this a bit so that the properties objects returns a dict of LLMArgs that will be passed here